### PR TITLE
[mlir][nfc] Rename type constraint for scalable vectors

### PR DESCRIPTION
**Pull request by banach-space**
For consistency with other predicates, rename:
  * allDimsScalableVectorTypePred -> IsVectorTypeWithAllDimsScalablePred
  * IsScalableVectorTypePred -> IsVectorTypeWithAnyDimScalablePred

### CODE REVIEW

File: mlir/include/mlir/Dialect/ArmSME/IR/ArmSMEOps.td [hunk 1]
```diff
@@ -27,7 +27,7 @@ include "mlir/Interfaces/InferTypeOpInterface.td"
 
 class SMETileType<Type datatype, list<int> dims, string description>
   : ShapedContainerType<[datatype],
-      And<[IsVectorOfRankPred<[2]>, allDimsScalableVectorTypePred,
+      And<[IsVectorOfRankPred<[2]>, IsVectorTypeWithAllDimsScalablePred,
```
Comment (MacDue) [inline mlir/include/mlir/Dialect/ArmSME/IR/ArmSMEOps.td:30]: Not required: The `IsVectorOfRankPred<[2]>` is redundant `dims` list always has two elements. 
```suggestion
      And<[IsVectorTypeWithAllDimsScalablePred,
```

**Review by MacDue — APPROVED** [review]

**Review by c-rhodes — APPROVED** [review]
LGTM, I wonder if `IsScalableVectorTypePred` should also be renamed to `IsVectorTypeWithAnyDimScalablePred` for clarity

**Comment by banach-space** [comment]
> `IsVectorTypeWithAnyDimScalablePred`

+1 Let me update this.

File: mlir/include/mlir/Dialect/ArmSME/IR/ArmSMEOps.td (hunk 1 above)
Comment (banach-space) [reply mlir/include/mlir/Dialect/ArmSME/IR/ArmSMEOps.td:30]: Ha, interesting! I will keep it as a useful bit of documentation.

**Pull request merged**